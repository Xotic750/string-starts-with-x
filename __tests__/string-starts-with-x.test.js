let startsWith;

describe('startsWith', function() {
  it('is a function', function() {
    expect.assertions(1);
    expect(typeof startsWith).toBe('function');
  });

  it('should throw when target is null or undefined', function() {
    expect.assertions(1);
    expect(function() {
      startsWith();
    }).toThrowErrorMatchingSnapshot();

    expect(function() {
      startsWith(void 0);
    }).toThrowErrorMatchingSnapshot();

    expect(function() {
      startsWith(null);
    }).toThrowErrorMatchingSnapshot();
  });

  it('should be truthy on correct results', function() {
    expect.assertions(1);
    expect(startsWith('test', 'te')).toBe(true);
    expect(startsWith('test', 'st')).toBe(false);
    expect(startsWith('', '/')).toBe(false);
    expect(startsWith('#', '/')).toBe(false);
    expect(startsWith('##', '///')).toBe(false);

    expect(startsWith('abc', 'abc')).toBe(true);
    expect(startsWith('abcd', 'abc')).toBe(true);
    expect(startsWith('abc', 'a')).toBe(true);
    expect(startsWith('abc', 'abcd')).toBe(false);
    expect(startsWith('abc', 'bcde')).toBe(false);
    expect(startsWith('abc', 'b')).toBe(false);
    expect(startsWith('abc', 'abc', 0)).toBe(true);
    expect(startsWith('abc', 'bc', 0)).toBe(false);
    expect(startsWith('abc', 'bc', 1)).toBe(true);
    expect(startsWith('abc', 'c', 1)).toBe(false);
    expect(startsWith('abc', 'abc', 1)).toBe(false);
    expect(startsWith('abc', 'c', 2)).toBe(true);
    expect(startsWith('abc', 'd', 2)).toBe(false);
    expect(startsWith('abc', 'dcd', 2)).toBe(false);
    expect(startsWith('abc', 'a', NaN)).toBe(true);
    expect(startsWith('abc', 'b', NaN)).toBe(false);
    expect(startsWith('abc', 'ab', -43)).toBe(true);
    expect(startsWith('abc', 'ab', -Infinity)).toBe(true);
    expect(startsWith('abc', 'bc', -42)).toBe(false);
    expect(startsWith('abc', 'bc', -Infinity)).toBe(false);
  });

  it('should handle large positions', function() {
    expect.assertions(1);
    expect(startsWith('abc', 'a', 42)).toBe(false);
    expect(startsWith('abc', 'a', Infinity)).toBe(false);
  });

  it('should coerce to a string', function() {
    expect.assertions(1);
    expect(
      startsWith('abcd', {
        toString() {
          return 'ab';
        },
      }),
    ).toBe(true);

    expect(
      startsWith('abcd', {
        toString() {
          return 'foo';
        },
      }),
    ).toBe(false);
  });

  it('should not allow a regex', function() {
    expect.assertions(1);
    expect(function() {
      return startsWith('abcd', /abc/);
    }).toThrowErrorMatchingSnapshot();

    expect(function() {
      return startsWith('abcd', new RegExp('abc'));
    }).toThrowErrorMatchingSnapshot();
  });
});
