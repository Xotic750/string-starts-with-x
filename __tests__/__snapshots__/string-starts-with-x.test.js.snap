// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`startsWith should not allow a regex 1`] = `"Cannot call method \\"startsWith\\" with a regex"`;

exports[`startsWith should not allow a regex 2`] = `"Cannot call method \\"startsWith\\" with a regex"`;

exports[`startsWith should throw when target is null or undefined 1`] = `"Cannot call method on undefined"`;

exports[`startsWith should throw when target is null or undefined 2`] = `"Cannot call method on undefined"`;

exports[`startsWith should throw when target is null or undefined 3`] = `"Cannot call method on null"`;
